### ISSUE TITLE
Test failure in OSGi build for 2.1

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/flow#8308]
https://bender.vaadin.com/viewLog.html?buildId=147233&tab=buildResultsDiv&buildTypeId=Flow_Nightly_FlowFelixJettyIntegrationTests21

`RouterSessionExpirationIT.navigationAfterSessionExpired` is failing.

**Comment by bogdanudrescu** [comment]
The test is successful on my computer. Still investigating.

**Comment by bogdanudrescu** [comment]
The build is using `bowerMode`.

I tried 
`mvn clean install`,
`mvn clean install -Pfelix-jetty` and 
`mvn clean install -Pfelix-jetty -Dvaadin.compatibilityMode=true`

The test always passes on my machine. There are other tests randomly failing but `RouterSessionExpirationIT` always passes.

Also I don't see yet the relation of the failure with a commit before the first failure which may be one of the two:
- Enable vaadin-prerelease for snapshot (#8165) 
- Push browser history state after navigation is certain, fixes parts of #3619

The failure is
```
Caused by: org.openqa.selenium.NoSuchElementException: no such element: Unable to locate element: {"method":"xpath","selector":"//strong[text()='ViewWhichInvalidatesSession']"}
```

This happen on `navigateToSesssionExpireView()` 
```java
        Assert.assertEquals(sessionId, getSessionId());
        navigateToSesssionExpireView();
```
which internally invokes
```
findElement(By.xpath("//strong[text()='" + linkText + "']"))
```

There is also a discrepancy between the behaviour of the automated test vs a manual test.
The automated test, after selecting `ViewWhichInvalidatesSession` waits for the selection to change, which change happens and selection remains on `ViewWhichInvalidatesSession`, until it's set to `NormalView` by `navigateToFirstView();`. 
This doesn't happen on a manual test, in which case `NormalView` is navigated directly by the invalid session in `ViewWhichInvalidatesSession`.
